### PR TITLE
backport(docs): add description to provider index page

### PR DESCRIPTION
**Pull request by matifali**
backports #271

### CODE REVIEW

**Comment by matifali** [comment]
@johnstcn any hints why integration tests are failing. This is a backport to `release/1.0` branch

File: docs/index.md [hunk 1]
```diff
@@ -3,11 +3,13 @@
 page_title: "Coder Provider"
 subcategory: "Infrastructure"
 description: |-
-  
+    Terraform provider for managing Coder templates, which are the underlying infrastructure for Coder workspaces.
```
Comment (bpmct) [inline docs/index.md:6]: Managing templates still may be a bit confusing as the coderd provider does too. Maybe we say something about "workspace internals"

Comment (matifali) [reply docs/index.md:6]: What if we call it `defining templates`?
```suggestion
    Terraform provider for defining/building Coder templates, which are the underlying infrastructure for Coder workspaces.
```

**Comment by johnstcn** [comment]
> @johnstcn any hints why integration tests are failing. This is a backport to `release/1.0` branch

you may need to cherry-pick https://github.com/coder/terraform-provider-coder/commit/5f660a0f7182975133fa768fcea67402e34b955c

File: docs/index.md (hunk 1 above)
Comment (johnstcn) [reply docs/index.md:6]: Let's backport this change first and iterate on the exact wording later.

File: docs/index.md [hunk 2]
```diff
@@ -3,11 +3,13 @@
 page_title: "Coder Provider"
 subcategory: "Infrastructure"
 description: |-
-  
+    Terraform provider for defining Coder templates, which are the underlying infrastructure for Coder workspaces.
 ---
 
 # Coder Provider
 
+Terraform provider for managing Coder [templates](https://coder.com/docs/templates), which are the underlying infrastructure for Coder [workspaces](https://coder.com/docs/workspaces).
```
Comment (matifali) [inline docs/index.md:11]:
```suggestion
Terraform provider for defining Coder [templates](https://coder.com/docs/templates), which are the underlying infrastructure for Coder [workspaces](https://coder.com/docs/workspaces).
```

File: templates/index.md.tmpl [hunk 3]
```diff
@@ -3,11 +3,13 @@
 page_title: "Coder Provider"
 subcategory: "Infrastructure"
 description: |-
-  
+    Terraform provider for managing Coder templates, which are the underlying infrastructure for Coder workspaces.
```
Comment (matifali) [inline templates/index.md.tmpl:6]:
```suggestion
    Terraform provider for defining Coder templates, which are the underlying infrastructure for Coder workspaces.
```

File: templates/index.md.tmpl [hunk 4]
```diff
@@ -3,11 +3,13 @@
...
 ---
 
 # Coder Provider
 
+Terraform provider for managing Coder [templates](https://coder.com/docs/templates), which are the underlying infrastructure for Coder [workspaces](https://coder.com/docs/workspaces).
```
Comment (matifali) [inline templates/index.md.tmpl:11]:
```suggestion
Terraform provider for defining Coder [templates](https://coder.com/docs/templates), which are the underlying infrastructure for Coder [workspaces](https://coder.com/docs/workspaces).
```

**Pull request merged**